Prepare Conversation Analysis SDK for 1.0 release

heaths opened the pull request.

heaths committed:
Prepare Conversation Analysis SDK for 1.0 release

heaths committed:
Add AAD support

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.AI.Language.Conversations](https://apiview.dev/Assemblies/Review/7521ad85a7df4f239d56c3c34700cc80)

heaths committed:
Add links to REST API documentation

Opened Azure/autorest#2384 to track support for defining links to REST API docs on each operation (method). The LLC generator supported this, but not the DPG generator.

heaths committed:
Update publish date in README

iscai-msft commented on sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Authoring/ConversationAuthoringClient.cs:
```diff
@@ -9,6 +9,10 @@
+    /// <remarks>
+    /// See <see href="https://docs.microsoft.com/rest/api/language/conversational-analysis-authoring"/> for more information about models you can pass to this client.
+    /// </remarks>
+    /// <seealso href="https://docs.microsoft.com/rest/api/language/conversational-analysis-authoring"/>
```
just want to make sure this renders only as one link, and isn't actually duplicated in docs

heaths replied:
Nope, no duplicate. The standalone `<seealso>` would appear in a separate "See Also" section like (from some random doc):

<img width="183" alt="image" src="https://user-images.githubusercontent.com/1532486/175649644-3b4d3fa1-aed3-4b76-89ad-41d556109f58.png">

iscai-msft commented on sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/autorest.md:
```diff
@@ -146,6 +149,25 @@ directive:
+  where: $.paths.*.*
+  transform: |
+    var operationId = $.operationId.substring("ConversationalAnalysisAuthoring_".length);
+    // BUGBUG: Will not work until https://github.com/Azure/autorest.csharp/issues/2384 is fixed.
```
nit: still says bug, idk if it's fixed

heaths replied:
Yep - this is the ol' school way we tracked outstanding bugs in code. I just opened the linked bug, so definitely not fixed yet. :) I wanted to keep this for posterity, if nothing else. Hopefully one day *soon* it will work.

iscai-msft reviewed (approved):
looks good!

heaths commented:
To note, this was regenerated on the DPG 1.0 GA merged with #29179. /cc @annelo-msft
